cluster-autoscaler/cloudprovider/azure/azure_template.go

```diff
@@ -36,46 +36,15 @@ import (
 
 const (
 	azureDiskTopologyKey string = "topology.disk.csi.azure.com/zone"
+	// AKSLabelPrefixValue represents the constant prefix for AKSLabelKeyPrefixValue
+	AKSLabelPrefixValue = "kubernetes.azure.com"
+	// AKSLabelKeyPrefixValue represents prefix for AKS Labels
+	AKSLabelKeyPrefixValue = AKSLabelPrefixValue + "/"
 )
 
-func buildInstanceOS(template compute.VirtualMachineScaleSet) string {
-	instanceOS := cloudprovider.DefaultOS
-	if template.VirtualMachineProfile != nil && template.VirtualMachineProfile.OsProfile != nil && template.VirtualMachineProfile.OsProfile.WindowsConfiguration != nil {
-		instanceOS = "windows"
-	}
-
-	return instanceOS
-}
-
-func buildGenericLabels(template compute.VirtualMachineScaleSet, nodeName string) map[string]string {
-	result := make(map[string]string)
-
-	result[apiv1.LabelArchStable] = cloudprovider.DefaultArch
-	result[apiv1.LabelOSStable] = buildInstanceOS(template)
-
-	result[apiv1.LabelInstanceTypeStable] = *template.Sku.Name
-	result[apiv1.LabelTopologyRegion] = strings.ToLower(*template.Location)
-
-	if template.Zones != nil && len(*template.Zones) > 0 {
-		failureDomains := make([]string, len(*template.Zones))
-		for k, v := range *template.Zones {
-			failureDomains[k] = strings.ToLower(*template.Location) + "-" + v
-		}
-
-		result[apiv1.LabelTopologyZone] = strings.Join(failureDomains[:], cloudvolume.LabelMultiZoneDelimiter)
-		result[azureDiskTopologyKey] = strings.Join(failureDomains[:], cloudvolume.LabelMultiZoneDelimiter)
-	} else {
-		result[apiv1.LabelTopologyZone] = "0"
-		result[azureDiskTopologyKey] = ""
-	}
-
-	result[apiv1.LabelHostname] = nodeName
-	return result
-}
-
-func buildNodeFromTemplate(scaleSetName string, template compute.VirtualMachineScaleSet, manager *AzureManager) (*apiv1.Node, error) {
+func buildNodeFromTemplate(nodeGroupName string, template compute.VirtualMachineScaleSet, manager *AzureManager, enableDynamicInstanceList bool) (*apiv1.Node, error) {
```

Comment:
The function definition in our fork is:
```
func buildNodeFromTemplate(nodeGroupName string, inputLabels map[string]string, inputTaints string,
	template compute.VirtualMachineScaleSet, manager *AzureManager, enableDynamicInstanceList bool) (*apiv1.Node, error) {
```

note: inputLabels and inputTaints are missing from the above

Reply:
That is intentional given those changes lead to core differences. The resolution for this will be done in another ticket.

Reply:
do we already have a ticket filed? if not, can we create one? and link this comment thread there so we don't forget